app/src/features/apiClient/screens/PostmanImporterView/components/PostmanImporter/PostmanImporter.tsx

```diff
@@ -0,0 +1,383 @@
+            try {
+              const fileContent = JSON.parse(reader.result as string);
+
+              if (!file.type.includes("json")) {
+                throw new Error("Invalid file. Please upload valid Postman export files.");
+              }
+
+              const fileType = getUploadedPostmanFileType(fileContent);
+              if (!fileType) {
+                throw new Error("Invalid file. Please upload valid Postman export files.");
+              }
+
+              if (fileType === "environment") {
+                const processedData = processPostmanEnvironmentData(fileContent);
+                resolve({ type: fileType, data: processedData });
+              } else {
+                const processedApiRecords = processPostmanCollectionData(fileContent);
+                const processedVariables = processPostmanVariablesData(fileContent);
+                resolve({
+                  type: fileType,
+                  data: {
+                    type: fileType,
+                    apiRecords: processedApiRecords,
+                    variables: processedVariables,
+                  },
+                });
+              }
+            } catch (error) {
+              Logger.error("Error processing postman file:", error);
+              reject(error);
+            }
+          };
+          reader.readAsText(file);
+        });
+      });
+
+      Promise.allSettled(processFiles)
+        .then((results) => {
+          const hasErrors = results.every((result) => result.status === "rejected");
+          if (hasErrors) {
+            throw new Error(
+              "Could not process the selected files!, Please check if the files are valid Postman export files."
+            );
+          }
+
+          const processedRecords: ProcessedData = {
+            environments: [],
+            apiRecords: [],
+            variables: {},
+          };
+
+          results.forEach((result: any) => {
+            if (result.status === "fulfilled") {
+              if (result.value.type === "environment") {
+                processedRecords.environments.push(result.value.data);
+              } else {
+                const { collections, apis } = result.value.data.apiRecords;
+                processedRecords.variables = { ...processedRecords.variables, ...result.value.data.variables };
+                processedRecords.apiRecords.push(...collections, ...apis);
+                collectionsCount.current += collections.length;
+              }
+            }
+          });
+
+          trackImportFromPostmanDataProcessed(collectionsCount.current, processedRecords.environments.length);
+
+          setProcessedFileData(processedRecords);
+          setProcessingStatus("processed");
+        })
+        .catch((error) => {
+          setImportError(error.message);
+          setProcessingStatus("idle");
+        })
+        .finally(() => {
+          if (importError) {
+            setProcessingStatus("idle");
+          }
+        });
```

Comment:
redundant `setProcessingStatus("idle")`
only the one in `.finally` seems enough